server/bleep/src/background/sync.rs

```diff
@@ -87,23 +87,46 @@ impl Drop for SyncHandle {
 }
 
 impl SyncHandle {
-    pub(crate) fn new(
+    pub(crate) async fn new(
         app: Application,
         reporef: RepoRef,
         status: super::ProgressStream,
         new_branch_filters: Option<crate::repo::BranchFilter>,
     ) -> Arc<Self> {
         let (exited, exit_signal) = flume::bounded(1);
-        let pipes = SyncPipes::new(reporef.clone(), status).into();
-        Self {
-            app,
+        let pipes = SyncPipes::new(reporef.clone(), status);
+        let sh = Self {
+            app: app.clone(),
+            reporef: reporef.clone(),
             pipes,
-            reporef,
             new_branch_filters,
             exited,
             exit_signal,
-        }
-        .into()
+        };
+
+        _ = sh
+            .create_new(|| {
```

Comment:
Can we just roll `create_new` into this method? Would make this logic easier to follow